_posts/2019-04-10-aws-dashboard-console-using-arduino.md

```diff
@@ -0,0 +1,265 @@
+
+After the response is read to the string, it is parsed by a JSON parser. I used the [ArduinoJSON](https://arduinojson.org/) library.
+
+**The physical LED lights and LCD screen content are set inside the `setContent` method:**
+
+```c++
+digitalWrite(PIPE_RUNNING_LED_DEV, LOW);
+digitalWrite(PIPE_FAILED_LED_DEV, LOW);
+digitalWrite(PIPE_RUNNING_LED_TEST, LOW);
+digitalWrite(PIPE_FAILED_LED_TEST, LOW);
+digitalWrite(ALARMS_LED, LOW);
+digitalWrite(PIPE_RUNNING_LED_PROD, LOW);
+digitalWrite(PIPE_FAILED_LED_PROD, LOW);
+
+if (pipelines_running_dev == true) { digitalWrite(PIPE_RUNNING_LED_DEV, HIGH); }
+if (pipelines_failed_dev == true) { digitalWrite(PIPE_FAILED_LED_DEV, HIGH); }
+if (pipelines_running_test == true) { digitalWrite(PIPE_RUNNING_LED_TEST, HIGH); }
+if (pipelines_failed_test == true) { digitalWrite(PIPE_FAILED_LED_TEST, HIGH); }
+if (alarms_raised_prod == true) { digitalWrite(ALARMS_LED, HIGH); }
+if (pipelines_running_prod == true) { digitalWrite(PIPE_RUNNING_LED_PROD, HIGH); }
+if (pipelines_failed_prod == true) { digitalWrite(PIPE_FAILED_LED_PROD, HIGH); } 
+```
+
+First, the LEDs are switch Off to clear the previous state. Then the necessary LEDs are switch On by the simple `if` clauses.
+
+**After the LED’s are fine, write content to the LCD:**
+
+```c++
+  for (int i=0; i < 5; i++) {
+
+    bool all_ok = true;
+  
+    if (pipelines_failed_dev == true) {
+      lcd.setCursor(0,0);
+      lcd.print("DEV Pipe fail:  ");
+      lcd.setCursor(0,1);
+      lcd.print(pipes_failed_0_dev);
+
+      all_ok = false;
+      delay(5000);
+    } 
+    
+    if (alarms_raised_prod == true) {  
+      lcd.setCursor(0,0);
+      lcd.print("PROD Alarms:    ");
+      lcd.setCursor(0,1);
+      lcd.print(alarms_0_prod);
+
+      all_ok = false;
+      delay(5000);
+    }
+
+    ...Other clauses...
+
+    if (all_ok == true) {  
+      lcd.clear();
+      lcd.print("     All OK     ");
+
+      delay(60000);
+
+      break;
+    }
+  }
+```
+
+Every `if` clause contains 5 second delay to keep the certain text on the screen for a moment. The `for` loop is used to “blink” all texts few times on the screen. If there are nothing to show, then print “All OK”.
+
+Few examples of the LCD output:
+
+![Dashboard LCD](/img/aws-dashboard-console-using-arduino/dashboard-lcd.png "Dashboard LCD")
+
+If the string is longer than 16 characters (LCD width), then it is cut. 
+
+>*The LCD library has an extra option to scroll overflowing content on the screen but I note that it was not suitable for this case. Scrolling every string on the screen would take just too much time.*
+
+### Lambda backend script
+AWS status information is served by the simple Lambda script (Python). The Lambda provides a REST interface that includes all necessary information for the console. The Lambda starts on request and does not contain persistence. The Lambda is publised though the AWS API Gateway and secured by an API KEY.
```

Comment:
publised -> published
though -> through

Reply:
necessary --> the necessary